pkg/kgo/source.go

```diff
@@ -1458,11 +1489,18 @@ func (a aborter) trackAbortedPID(producerID int64) {
 // processing records to fetch part //
 //////////////////////////////////////
 
+var rawRecordsPool = pool.NewBucketedPool[kmsg.Record](32, 4096, 2, func(len int) []kmsg.Record {
```

Comment:
also on the size: we'd go as high as 6800 records per fetch response. Maybe something like 16Ki is a more reasonable max size?

Reply:
_addressed @ 4eed8447f29fb192f361cc04d28fccc3e450ba85_